sarsCoV2Analysis.wdl

```diff
@@ -0,0 +1,472 @@
+    File ampliconBed
+    String sample
+    Int mem = 8
+    Int timeout = 72
+  }
+
+  String primertrim = "~{sample}.primertrim"
+  String primertrimBam = "~{sample}.primertrim.bam"
+  String sortedBam_ = "~{sample}.primertrim.sorted.bam"
+  String sortedBai_ = "~{sample}.primertrim.sorted.bam.bai"
+
+  command <<<
+    set -euo pipefail
+
+    ivar trim -i ~{bam} -b ~{primerBed} -p ~{primertrim}
+
+    samtools sort ~{primertrimBam} -o ~{sortedBam_}
+
+    samtools index ~{sortedBam_}
+
+    bedtools coverage -hist -a ~{ampliconBed} -b ~{sortedBam_} > ~{sample}.cvghist.txt
+  >>>
+
+  runtime {
+    memory: "~{mem} GB"
+    modules: "~{modules}"
+    timeout: "~{timeout}"
+  }
+
+  output {
+    File? sortedBam = "~{sortedBam_}"
+    File? sortedBai = "~{sortedBai_}"
+    File? cvgHist = "~{sample}.cvghist.txt"
+  }
+}
+
+task variantCalling {
+  input {
+    String modules = "bcftools/1.9 samtools/1.9 vcftools/0.1.16 seqtk/1.3 sars-covid-2-bowtie-index/2.3.5.1 sars-covid-2/mn908947.3"
+    File bam
+    String sample
+    String sarsCovidRef = "$SARS_COVID_2_ROOT/MN908947.3.fasta"
+    Int mem = 8
+    Int timeout = 72
+  }
+
+  String vcfName = "~{sample}.vcf"
+  String fastaName = "~{sample}.consensus.fasta"
+  String variantOnlyVcf_ = "~{sample}.v.vcf"
+
+  command <<<
+    set -euo pipefail
+
+    samtools mpileup -aa -uf ~{sarsCovidRef} ~{bam} | \
+    bcftools call --ploidy 1 -Mc | tee -a ~{vcfName} | \
+    vcfutils.pl vcf2fq -d 10 | \
+    seqtk seq -A - | sed '2~2s/[actg]/N/g' > ~{fastaName}
+
+    bcftools mpileup -a "INFO/AD,FORMAT/DP,FORMAT/AD" \
+    -d 8000 -f ~{sarsCovidRef} ~{bam} | \
+    tee ~{sample}.m.vcf | bcftools call --ploidy 1 -m -v > ~{variantOnlyVcf_}
+  >>>
+
+  runtime {
+    memory: "~{mem} GB"
+    modules: "~{modules}"
+    timeout: "~{timeout}"
+  }
+
+  output {
+    File vcfFile = "~{vcfName}"
+    File consensusFasta = "~{fastaName}"
+    File variantOnlyVcf = "~{variantOnlyVcf_}"
+  }
+}
+
+task qcStats {
+  input {
+    String modules = "bedtools samtools/1.9"
+    String sample
+    File bam
+    File hostMappedBam
+    Int mem = 8
+    Int timeout = 72
+  }
+
+  command <<<
+    set -euo pipefail
+
+    bedtools genomecov -ibam ~{bam} > ~{sample}.genomecvghist.txt
+
+    bedtools genomecov -d -ibam ~{bam} > ~{sample}.genome.cvgperbase.txt
+
+    samtools stats ~{hostMappedBam} > ~{sample}.host.mapped.samstats.txt
+
+    samtools stats ~{bam} > ~{sample}.samstats.txt
+  >>>
+
+  runtime {
+    memory: "~{mem} GB"
+    modules: "~{modules}"
+    timeout: "~{timeout}"
+  }
+
+  output {
+    File genomecvgHist = "~{sample}.genomecvghist.txt"
+    File genomecvgPerBase = "~{sample}.genome.cvgperbase.txt"
+    File hostMappedAlignmentStats = "~{sample}.host.mapped.samstats.txt"
+    File hostDepletedAlignmentStats = "~{sample}.samstats.txt"
+  }
+}
+
+task blast2ReferenceSequence {
+  input {
+    String modules = "blast"
+    String reference = "/.mounts/labs/gsiprojects/gsi/covid19/ref/MN908947.3.fasta"
```

Comment:
Better to have:
`module load sars-covid-2/mn908947.3`
`String reference = "$SARS_COVID_2_ROOT/MN908947.3.fasta"`